modules/tinymce/src/core/test/ts/browser/EditorCleanupTest.ts

```diff
@@ -0,0 +1,47 @@
+import { Assertions, Chain, Log, NamedChain, Pipeline } from '@ephox/agar';
+import { Editor as McEditor } from '@ephox/mcagar';
+import { UnitTest } from '@ephox/bedrock-client';
+import { Attr, Element, Truncate } from '@ephox/sugar';
+import { HTMLDivElement } from '@ephox/dom-globals';
+
+import Editor from 'tinymce/core/api/Editor';
+import { RawEditorSettings } from 'tinymce/core/api/SettingsTypes';
+import Theme from 'tinymce/themes/silver/Theme';
+import VisualBlocksPlugin from 'tinymce/plugins/visualblocks/Plugin';
+
+UnitTest.asynctest('browser.tinymce.core.EditorCleanupTest', (success, failure) => {
+  Theme();
+  VisualBlocksPlugin();
+
+  const base_url = '/project/tinymce/js/tinymce';
+
+  const cTestCleanup = (settings: RawEditorSettings, html: string = '<div></div>') => NamedChain.asChain([
+    NamedChain.direct(NamedChain.inputName(), McEditor.cFromHtml(html, settings), 'editor'),
+    NamedChain.direct('editor', Chain.mapper((editor: Editor) => Element.fromDom(editor.getElement())), 'element'),
+    NamedChain.read('editor', Chain.op((editor: Editor) => editor.remove())),
+    // first, remove the id of the element, as that's inserted from McEditor.cFromHtml and is out of our control
+    NamedChain.read('element', Chain.op((element: Element<HTMLDivElement>) => Attr.remove(element, 'id'))),
+    NamedChain.direct('element', Chain.mapper(Truncate.getHtml), 'raw-html'),
+    NamedChain.read('raw-html', Assertions.cAssertHtml('all properties on the element should be cleaned up', html)),
+    NamedChain.read('editor', McEditor.cRemove)
```

Comment:
This won't work, as it relies on looking up the element based on the id that you removed above. Ideally, you should just call this above instead of `editor.remove()`.

Reply:
For some reason that fix leaves the element that I'm testing with on the screen. I've swapped `McEditor.cRemove` for `Chain.op(Remove.remove)` on that last line and that's fixed the bug

Reply:
As discussed on slack, this works but would be better if we stick to using the `McEditor.cRemove` chain. Reason being there's normally additional content that should be cleaned up, since TinyMCE doesn't do that atm (eg stylesheets). In saying that though, McEditor seems to not be doing that which is a bug.

Here's an example of what worked:
```
  const cTestCleanup = (settings: RawEditorSettings, html: string = '<div></div>') => NamedChain.asChain([
    // spin the editor up and down, getting a reference to its target element inbetween
    NamedChain.write('editor', McEditor.cFromHtml(html, { base_url: '/project/tinymce/js/tinymce', ...settings })),
    NamedChain.direct('editor', Chain.mapper((editor: Editor) => Element.fromDom(editor.getElement())), 'element'),
    NamedChain.read('editor', McEditor.cRemove),
    // first, remove the id of the element, as that's inserted from McEditor.cFromHtml and is out of our control
    NamedChain.read('element', Chain.op((element: Element<HTMLDivElement>) => Attr.remove(element, 'id'))),
    // assert that the html of the element is correct
    NamedChain.direct('element', Chain.mapper(Truncate.getHtml), 'raw-html'),
    NamedChain.read('raw-html', Assertions.cAssertHtml('all properties on the element should be cleaned up', html))
  ]);
```